common/lib/common-utils/api-report/common-utils.api.md

```diff
@@ -15,6 +15,7 @@ import { ITelemetryErrorEvent } from '@fluidframework/common-definitions';
 import { ITelemetryGenericEvent } from '@fluidframework/common-definitions';
 import { ITelemetryLogger } from '@fluidframework/common-definitions';
 import { ITelemetryPerformanceEvent } from '@fluidframework/common-definitions';
+import { ReplaceIEventThisPlaceHolder } from '@fluidframework/common-definitions';
 import { TransformedEvent } from '@fluidframework/common-definitions';
```

Comment:
shouldn't this change be in next so you can actually consume it?

Reply:
Yeah I realized that as well.  Was waiting for someone else to confirm before shuffling.  Now that I think of it, I didn't get any type validation errors to annotate... So let me look into a bit more, maybe it's not technically breaking somehow...?

Reply:
Ok yeah this reminds me of a very curious - and backwards to me - thing I observed about TypeScript type constraints when extending a class.  An overload of a function may restrict the paramater types but not relax them.  This makes no sense to me but that's what's going on here and I confirmed it with a toy example in TS Playground.  I'll go find the link

Reply:
Playground example showing derived class can restrict the parameter type (which is what I'm doing too): [TS Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBANiA5jAvDUkRwKYDoGIDcAUMRtDGAajABRgBclArgLYBG2ATgJSoB8dAB5MAhmACefFIPJY8BWgAMA2gBIA3mAC+AXRiah2pTxJk4oiBBgAhS9hgbiMFzEQha2AG7YwUJtBcAJZgyAA+LBzcfE6ucZSKAAw8nj5+ps6u2sTZ5pbWACLcQT4AJjDYQlC+pdZ2EA6xru6pvv4wgSGIMTAA9L2RnFyUILCicAgA7tjlABbc2JnxHcwADty4Ld5tGcsuVEi0AIwp2+kkex2TQVDAs3RnUDFLl+j2MABE8xMgH0wHiGOKQ+wHsRw+phg7C42FEAGsLq8XKCGp93CBSuwJNg-glDidaCD7AAmCGEKEw+GIpGlbAAM1EzDg-heSP2igJHwAqgA5ACiAA0AAp8gDCABU+QUPjAANQVNJPanxbJxbK5ciwcpoMDYSYwIrBMq0DKamClSw2GjlSy2exmRQZUqbDxfbA-MnER0kfowAACUAgAFpKutgFAQ1wuCAuMRnS0jsSAMwZb1xy0u2gAVgALKnDhkgA)

Reply:
Anyway, type validation/compatibility aside, this will cause breaks if someone has the wrong args for `emit` - my tests demonstrate that.  So yeah, in the final analysis it belongs in `next`.  Agreed @anthony-murphy ?

PS - As I try different signatures for the function in the example above I get surprising results depending on the types I'm trying. So there must be more nuance to the type compatibility here than I'm grasping.  Not really necessary to understand it all the way I guess.

Reply:
Also, I think I'll have to consume it in next as well, since Aqueduct DataObject ultimately extends TypedEventEmitter.

Now, it actually extends `EventForwarder` which extends `TypedEventEmitter` - Not sure if calling `emit` on an EventForwarder directly is expected or not?  Kind of a moot point since it's possible therefore needs to be supported - if not expected could be something to consider tightening up.

Reply:
The reason you don't see breaks in the type tests is because the class is generic, so by default it is just picking IEvent, which happens to match the previous signature of emit, but as you call out, other generic will not

Reply:
Oh good point. Actually it passes `any` (there's no default).  But... even with a specific type like `{ (event: "foo", listener: () => void)}` it's equivalent.

Reply:
a couple other things, have you consumed this in the repo manually? either via symlink, or just putting they type somewhere like core-interfaces, fixing up refs, and building? On that note, do we want to perpetuate the existing pattern? Can we now build a better map-like pattern? if so, did you consider just forking? and specifically forking directly into the client release group, rather than keeping it in common-utils?

Reply:
Great questions - Let's discuss at Office Hours :)